### DISCUSSION TITLE
Specify type based on include parameter

### DISCUSSION THREAD

**Discussion by Vallo** [prisma/prisma#12697] [Q&A]
I have an entity with many relations and I want a function to handle the include property as parameter:

```
export const findById = async (id: number, include: Prisma.ClaimInclude | undefined = undefined) => {
    return dbClient.claim.findUnique({
        where: {
            id,
        },
        include,
    })
}
```

so I would use this function like this

`const claim = await findById(claimId , { assignedTo: true, })`

my problem is that now I cannot access the assignedTo property

> Property 'assignedTo' does not exist on type 'Claim & {}'. Did you mean 'assignedToId'? ts(2551)

is there a way I could make this approach work, or do I need to have specific functions for each include type? 

Thanks

**Comment by Vallo** [comment]
found my solution here https://github.com/prisma/prisma/issues/3372

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
